File: src/main/scala/com/abaddon83/cardsGames/burracoGames/domainModels/burracoGames/completed/PlayerScore.scala
package com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.completed

import com.abaddon83.cardsGames.burracoGames.domainModels.BurracoPlayer
import com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.initialised.playerInGames.{Burraco, PlayerInGame}
import com.abaddon83.cardsGames.shares.decks.Card
import com.abaddon83.cardsGames.shares.players.PlayerIdentity

case class PlayerScore(
                                playerIdentity: PlayerIdentity,
                                winner: Boolean,
                                burracoList: List[BurracoPoint],
                                remainedCards: List[Card]
                              ) extends BurracoPlayer {

}

object PlayerScore{
  def build(player: PlayerInGame,winner: Boolean): PlayerScore = {
    PlayerScore(
    playerIdentity = player.playerIdentity,
    winner = winner,
    burracoList = burracoList(player.burracoList()),
    remainedCards = player.showMyCards()
    )
  }

  def burracoList(burracoList: List[Burraco]): List[BurracoPoint] ={
    burracoList.map(burraco =>
      BurracoPoint(burraco)
    )
  }
}

File: src/main/kotlin/com/abaddon83/burracoGame/domainModels/burracoGameendeds/PlayerScore.kt
package com.abaddon83.burracoGame.domainModels.burracoGameendeds

import com.abaddon83.burracoGame.domainModels.BurracoPlayer
import com.abaddon83.burracoGame.domainModels.burracoGameExecutions.playerInGames.PlayerInGame
import com.abaddon83.burracoGame.shared.burracos.Burraco
import com.abaddon83.burracoGame.shared.decks.Card
import com.abaddon83.burracoGame.shared.players.PlayerIdentity

data class PlayerScore private constructor(
        override val identity: PlayerIdentity,
        val winner: Boolean,
        val burracoList: List<BurracoPoint>,
        val remainedCards: List<Card>
): BurracoPlayer() {

    companion object Factory{
        fun create(player: PlayerInGame, winner: Boolean): PlayerScore =
            PlayerScore(
                    identity = player.identity(),
                    winner = winner,
                    burracoList = burracoList(player.burracoList()),
                    remainedCards = player.showMyCards()
            )

        private fun burracoList(burracoList: List<Burraco>): List<BurracoPoint> =
            burracoList.map { burraco ->
                BurracoPoint(burraco)
            }

    }

}